### PR TITLE
Update instructions for building examples

### PR DESCRIPTION
**Pull request by oyvindnetland**
Hi
Trying a relatively simple change here to test out contributing to this repo. 
As mentioned in https://github.com/esp-rs/esp-idf-hal/issues/270 I had some issues building the repo, so I wanted to update the README.md to make it abit easier.

Also related to https://github.com/esp-rs/esp-idf-hal/issues/240

Thanks to @Vollbrecht for helping out with how the espflash commands work for different versions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,8 +40,15 @@ _\* When using Octal Flash and/or Octal PSRAM_
...
+
+with cargo-esptool v1.7:
+```
+$ ESP_IDF_VERSION=release/v4.4 cargo espflash --target xtensa-esp32-espidf --example ledc_simple --monitor /dev/ttyUSB0
+```
+
+with cargo-esptool v2.0:
 ```
-$ cargo espflash --release --target riscv32imc-esp-espidf --example ledc_simple --monitor /dev/ttyUSB0
+$ ESP_IDF_VERSION=release/v4.4 cargo espflash flash --target xtensa-esp32-espidf --example ledc_simple --monitor
```
Comment (ivmarkov) [inline README.md:51]: The target for `ESP32-C3` (which is mentioned above) *must* be `riscv32imc-esp-espidf` so if you can restore that.

Also you might want to replace "To run `ledc_simple` on an ESP32-C3" with "To run e.g. `ledc_simple` e.g. on an ESP32-C3" to make it clear, that the same command line (modulo the target name and the example name) can be used for other MCUs and for other examples.

And you can add

**Review by ivmarkov — CHANGES_REQUESTED** [review]

**Comment by ivmarkov** [comment]
Regarding #241, I really would not like to repeat the "how to install all Rust tooling for ESP-RS" in all `esp-idf-*` crates given that we have a good section about this [here](https://github.com/esp-rs/esp-idf-template#prerequisites), and we should refer to that section rather than copy-paste it.

@Vollbrecht agreed to this but then somehow this effort stalled. If he does not have time, I might take his suggestion, and simplify it as per the requirement from above ^^^.

File: README.md (hunk 1 above)
Comment (oyvindnetland) [reply README.md]: done

**Comment by oyvindnetland** [comment]
> Regarding #241, I really would not like to repeat the "how to install all Rust tooling for ESP-RS" in all `esp-idf-*` crates given that we have a good section about this [here](https://github.com/esp-rs/esp-idf-template#prerequisites), and we should refer to that section rather than copy-paste it.
> 
> @Vollbrecht agreed to this but then somehow this effort stalled. If he does not have time, I might take his suggestion, and simplify it as per the requirement from above ^^^.

If i could offer the perspective of a "new user" of esp-rs/* then i would say that it is abit confusing that some of the same howto information is repeated several times, possibly with slight variations. Which is why i just jumped in an tried to at least do some corrections to make this slightly easier. 
But i did not catch the PR you mention that tries to do some of the same. But i think that lack the mention of how to build examples, which is a neat, but not obvious cargo feature.

**Pull request merged**